<launch>
    <!-- set logging level -->
    <param name="log_level" value="0"/> <!-- DEBUG=0, INFO=1, WARN=2, ERROR=3, FATAL=4 -->
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find logcart_driver)/config/custom_rosconsole.conf" />

    <!-- remap topic namespace -->
    <remap from="/joystick/cmd_vel" to="/base_controller/cmd_vel" />
    <remap from="/tracking/cmd_vel" to="/base_controller/cmd_vel" />
    <remap from="odom" to="/base_controller/odom" />

    <!-- joy node -->
    <node name="joy_node" pkg="joy" type="joy_node" respawn="true">
        <param name="dev" type="string" value="/dev/input/js1" />
        <param name="deadzone" value="0.25" />
        <param name="autorepeat_rate" value="1" /> <!-- in Hz -->
    </node>

    <!-- teleop node -->
    <node name="teleop_joy_node" pkg="logcart_driver" type="teleop_joy" output="screen">
        <param name="axis_linear" value="1" type="int" />
        <param name="axis_angular" value="2" type="int" />
        <param name="scale_linear" value="1.0" type="double" />
        <param name="scale_angular" value="1.0" type="double" />
    </node>
    <!-- mbed hardware interface -->
    <node name="mbed_node" pkg="logcart_driver" type="mbed_interface" output="screen" respawn="true">
        <param name="port_mbed" type="string" value="/dev/ttyACM0" />
        <param name="baud_rate" type="int" value="115200" /> -->
        <!-- <param name="port_mbed" type="string" value="/dev/tnt0" /> -->
    </node>

    <!-- ROS control node -->
    <rosparam command="load" file="$(find logcart_driver)/config/logcart_controller.yaml" />
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="true" output = "screen" args="base_controller" />


</launch>
